packages/ef-runtime-client/src/ComponentRegistry/ComponentRegistry.ts

```diff
@@ -12,35 +14,66 @@ export interface IComponentRegistryDependencies {
 
 export class ComponentRegistry implements IComponentRegistry {
   private registry: { [propName: string]: string } = {};
+  private initialised: boolean;
   private registryURL: string;
 
   constructor({ registryURL }: IComponentRegistryDependencies) {
     this.registryURL = registryURL;
+    this.initialised = false;
   }
 
   async fetch(systemCode: string): Promise<void> {
     try {
       const res = await fetch(`${this.registryURL}/?app=${systemCode}`);
       const data = await res.json();
       Object.assign(this.registry, data.imports);
+      this.initialised = true;
     } catch (err) {
-      console.error("Unable to fetch Component Registry", err);
+      logger.error("Unable to fetch Component Registry", err);
     }
   }
 
   getURL(component: string): string | undefined {
+    if (!this.initialised) {
+      logger.warn(
+        "Unable to get component URL. Component Registry was not initialised yet"
+      );
+      return;
+    }
     return this.registry[component];
   }
 
   getComponentKeys(): string[] {
+    if (!this.initialised) {
+      logger.warn(
+        "Unable to get component keys. Component Registry was not initialised yet"
+      );
+      return;
+    }
     return Object.keys(this.registry);
   }
 
   getRegistry(): { [key: string]: string } {
+    if (!this.initialised) {
+      logger.warn(
+        "Unable to get Registry. Component Registry was not initialised yet"
+      );
+      return {};
+    }
     return { ...this.registry };
   }
 
   applyOverrides(overrides: { [propName: string]: string }): void {
-    Object.assign(this.registry, overrides);
+    try {
+      Object.assign(this.registry, overrides);
+      if (Object.keys(overrides).length > 0) {
+        // if override includes at least one entry
+        // considers the components initialised even
+        // if fetch was not called
+        this.initialised = true;
```

Comment:
I don't think we need to set `this.initialised` to true here. it's enough to set it in the `fetch` method. Besides, fetch could have been called and the registry that it returns can possibly be empty.